core/modules/util/EventThread.cc

```diff
@@ -0,0 +1,164 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2015 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ *
+ *  @author: John Gates,
+ */
+
+// Class header
+#include "util/EventThread.h"
+
+// System headers
+#include <algorithm>
+
+// Third-party headers
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+namespace lsst {
+namespace qserv {
+namespace util {
+
+/// Handle commands as they arrive until queEnd() is called.
+void EventThread::handleCmds() {
+    startup();
+    while(_loop) {
+        Command::Ptr cmd = _q->getCmd();
+        if (cmd != nullptr) {
+            _q->commandStart(cmd);
+            if (cmd->action() == HALT) {
+                _loop = false;
+            }
+            _q->commandFinish(cmd);
+            cmd->resetFunc(); // Reset _func in case it has a captured Command::Ptr
+        }
+    }
+    finishup();
+}
+
+/// call this to start the thread
+void EventThread::run() {
+    std::thread t{&EventThread::handleCmds, this};
+    _t = move(t);
```

Comment:
@abh3 I think the rule is "abso not in header files, but you can in inplementation files".  In any case, there's a std::thread right above here, so consistency, etc.?